code-experiments/build/python/example_experiment_non_anytime.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python
+"""An example experiment for benchmarking non-anytime optimization algorithms with restarts.
+
+Non-anytime algorithms are those whose parameters depend not only on the dimensionality of the problem, but also on the
+budget of evaluations. Benchmarking such algorithms therefore entails running experiments on a test suite using various
+(increasing) budgets. These budgets can/should be provided by the user. If the user does not specify a budget list, a
+default budget list is used.
+
+This code builds upon the example experiment for beginners and provides only simplistic progress information.
+
+To apply the code to a different solver, add an interface to function `fmin`.
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+import cocoex  # experimentation module
+from numpy.random import rand  # for randomised restarts
+
+# for postprocessing
+import cocopp
+import os
+import webbrowser
+
+# for solvers
+import scipy.optimize
+import cma
+
+
+def default_budget_list(max_budget=10, num=100):
+    """Produces a budget list with at most `num` different increasing budgets within [1, `max_budget`] that are equally
```

Comment:
Python lines should generally not be longer than 80 characters see https://www.python.org/dev/peps/pep-0008/#maximum-line-length. Same for above docstring, 80-character lines are nicer to read anyways.

Reply:
True! I was mislead by PyCharm... (https://stackoverflow.com/questions/26808681/why-does-pycharm-use-120-character-lines-even-though-pep8-specifies-79)

The PEP 8 guide is actually stricter: *Limit all lines to a maximum of **79 characters**. 
For flowing long blocks of text with fewer structural restrictions (docstrings or comments), the line length should be limited to **72 characters**.*

I'm now using a limit of 79 characters for everything, but the code is less readable now (in my opinion, of course). I can shorten the docstrings/comments to 72 characters, if you think it's necessary.

BTW, the guide says also: *For code maintained exclusively or primarily by a team that can reach agreement on this issue, it is okay to increase the nominal line length from 80 to **100 characters** (effectively increasing the maximum length to 99 characters), provided that comments and docstrings are still wrapped at 72 characters.* Just a hint ;-)